### PR TITLE
Enable the context priority setting.

### PR DESCRIPTION
**Pull request by hongbolv**

### CODE REVIEW

**Comment by hongbolv** [comment]
> I can't find where we report max value of supported priority to application. Could you point to this place?

This PR is to set the context priority to Linux kmd. We have not implemented the priority report yet. The VAAPI prototype is https://github.com/intel/libva/pull/261.

**Comment by hongbolv** [comment]
> I can't find where we report max value of supported priority to application. Could you point to this place?

Updated, the max value is defined as 1024 in the media driver.

**Comment by FurongZhang** [comment]
Just want to confirm, could we change context priority dynamically? After context creation, is it possible to change the priority? Thanks.

**Comment by hongbolv** [comment]
> Just want to confirm, could we change context priority dynamically? After context creation, is it possible to change the priority? Thanks.

It doesn't support context priority dynamically change. The priority is only configured through context creation.

**Comment by FurongZhang** [comment]
Approved for VP code changes.

**Comment by XinfengZhang** [comment]
hi @hongbolv , 1, please add VA VERSION CHECK , refer https://github.com/intel/media-driver/pull/1012
2. if possible , could you split it to several commit? such as: step1. MOS change, step2 DDI set change, step 3. Capability change?.

File: media_driver/linux/common/ddi/media_libva_caps.cpp [hunk 1]
```diff
@@ -1960,6 +1991,7 @@ VAStatus MediaLibvaCaps::CreateDecConfig(
                 && decAttributes[1].value == m_decConfigs[i].m_encryptType
                 && decAttributes[2].value == m_decConfigs[i].m_processType)
         {
+            m_decConfigs[i].m_priority = decAttributes[3].value - CONTEXT_PRIORITY_MAX/2;
```
Comment (XinfengZhang) [inline media_driver/linux/common/ddi/media_libva_caps.cpp:1994]: need to add check about decAttributes[3].value. is it valid.

File: media_driver/linux/common/ddi/media_libva_caps.cpp [hunk 2]
```diff
@@ -2046,6 +2079,10 @@ VAStatus MediaLibvaCaps::CreateEncConfig(
                 return VA_STATUS_ERROR_UNSUPPORTED_RT_FORMAT;
             }
         }
+        if(VAConfigAttribContextPriority == attribList[j].type)
+        {
+            priority = attribList[j].value - CONTEXT_PRIORITY_MAX/2;
```
Comment (XinfengZhang) [inline media_driver/linux/common/ddi/media_libva_caps.cpp:2084]: same

File: media_driver/linux/common/ddi/media_libva_caps.cpp (hunk 1 above)
Comment (hongbolv) [reply media_driver/linux/common/ddi/media_libva_caps.cpp:1994]: > hi @hongbolv , 1, please add VA VERSION CHECK , refer #1012
Done in the PR:https://github.com/intel/media-driver/pull/1020
> 2. if possible , could you split it to several commit? such as: step1. MOS change, step2 DDI set change, step 3. Capability change?.
Done in the PR: https://github.com/intel/media-driver/pull/1020

File: media_driver/linux/common/ddi/media_libva_caps.cpp (hunk 2 above)
Comment (hongbolv) [reply media_driver/linux/common/ddi/media_libva_caps.cpp:2084]: Done the PR: https://github.com/intel/media-driver/pull/1020

**Comment by hongbolv** [comment]
@XinfengZhang  I have split the PR into https://github.com/intel/media-driver/pull/1019 and https://github.com/intel/media-driver/pull/1020. Please help have a review.

**Comment by hongbolv** [comment]
Changed to the dynamic session priority implementation.

**Pull request closed without merging**